dts/bindings/pinctrl/nxp,s32k3-pinctrl.yaml

```diff
@@ -0,0 +1,114 @@
+# Copyright 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
```

Comment:
Why introduce a new pin control compatible, if we are reusing a driver? If the driver can be common between multiple S32 parts, I would rather see the compatible reused (or renamed to be common between the SOCs, since it will only be used at the SOC level).

Reply:
The reason is that the features supported for each family of devices is different, even if they use the same driver. Perhaps in the future we can look to another approach to represent the pinctrl supported features and reset values per pin, as is done for some of the NXP parts (i.mx I believe) representing this with DT nodes in the HAL.

Reply:
I've rename the binding and reuse the driver for all NXP S32 family devices.

Reply:
I have reworked the pinctrl_soc.h headers and unify them by using the HAL macros that expose the features supported on the devices. I still keep different bindings for each family of devices (Z/E, K3) to expose different DT properties following the the features and allowed values particular to each device, as it's done also for other NXP parts (like i.mx) and vendors.